contracts/sources/obligation.move

```diff
@@ -167,6 +184,46 @@ module suilend::obligation {
         }
     }
 
+    public(friend) fun set_emode<P>(
+        obligation: &mut Obligation<P>,
+        deposit_reserve: &Reserve<P>,
+        borrow_reserve: &Reserve<P>,
+    ) {
+        assert!(!is_emode(obligation), EEModeAlreadySet);
+        assert!(vector::length(&obligation.borrows) <= 1, EEModeNotValidWithCrossMargin);
+        assert!(vector::length(&obligation.deposits) <= 1, EEModeNotValidWithCrossMargin);
+
+        let deposit_reserve_array_index = reserve::array_index(deposit_reserve);
+        let borrow_reserve_array_index = reserve::array_index(borrow_reserve);
+
+        let is_valid_emode = reserve_config::check_emode_validity(
+            config(deposit_reserve), &borrow_reserve_array_index
+        );
+        assert!(is_valid_emode, EEmodeInvalidForPairProvided);
+
+        // The obligation must ONLY have one deposit and one borrow, and
+        // they both must match the emod ones
+        if (vector::length(&obligation.borrows) == 1) {
```

Comment:
hmm. so the obligation has to have a deposit and borrow before activating e-mode? This feels a bit restrictive. IMO all the states below are valid:
- has no deposit, no borrow
- has deposit, no borrow
- has deposit, borrow

Reply:
changed with the last iteration